zerver/lib/emoji_utils.py

```diff
@@ -0,0 +1,22 @@
+# This file doesn't import from django so that we can use it in `build_emoji`
+import re
+import unicodedata
+
+
+def unqualify_emoji(emoji: str) -> str:
+    # Not allowing \u20e3 excludes keycaps, which we leave qualified
+    # so that if the user types an unqualified keycap, we don't turn
+    # it into an image.
```

Comment:
Similarly this would benefit from a docstring explaining what "unqualifying an emoji" means.

Reply:
I moved most of the other comment here, so that the more in depth explanation happens in the function with `unqualify` in its name